alerts/kubelet.libsonnet

```diff
@@ -24,7 +24,7 @@
           },
           {
             expr: |||
-              kube_node_spec_taint{%(kubeStateMetricsSelector)s,key="node.kubernetes.io/unreachable",effect="NoSchedule"} == 1
+              kube_node_spec_taint{%(kubeStateMetricsSelector)s,key="node.kubernetes.io/unreachable",effect="NoSchedule"} - kube_node_spec_taint{%(kubeStateMetricsSelector)s,key="ToBeDeletedByClusterAutoscaler",effect="NoSchedule"} == 1
```

Comment:
Does this actually work? I would have expected to see an `on(node)` statement in there.

Reply:
And probably `topk by(node) (1, <statement>)`, for safety with mutliple kube-state-metrics instances on both sides.

Reply:
Yes, it Works. Although que promql showed it, I did wait until our dev environment went down on the afternoon before opening this.  
About using _on_, you're right but is very unlikely that node names mismatch. Anyway looking to another operator bundled alerts (not sure wheter from prometheus-mixin or anywhere else) I don't see this kind of subtle considerations. Specifically KubeJobCompletion, whith query `sum ( kube_job_spec_completions{job="kube-state-metrics",namespace=~".*"} - kube_job_status_succeeded{job="kube-state-metrics",namespace=~".*"} ) > 0`

Reply:
It's not required in the kube job completion one as both sides have the same labelset, which is not the case in this query. Could you share a screenshot of the result you describe?

Reply:
This is a very minor fix for so much questions. If you dislike or believe is useless, just close the pull request.  
I've applied the fix on my instances and are very happy with it. I'll not work more on this from my side.

Reply:
A change being one line, doesn't necessarily make it simple. Let me explain why I don't think it works as expected.

Operator arithmetic: In this situation we are using an operator where both sides are vectors. See ["Between two instant vectors"](https://prometheus.io/docs/prometheus/latest/querying/operators/#arithmetic-binary-operators), it mentions [vector matching](https://prometheus.io/docs/prometheus/latest/querying/operators/#vector-matching). Vector matching, describes [one-to-one](https://prometheus.io/docs/prometheus/latest/querying/operators/#one-to-one-vector-matches) matching, where both side _must_ have the same labelset. If there are no vectors on each side, that have a matching labelset, then the result set will be empty. The query you proposed specifically mentions `key="node.kubernetes.io/unreachable"` on the left hand side, and `key="ToBeDeletedByClusterAutoscaler"` on the right hand side, the labelsets can never match, thus the result will always be empty, and will never alert you.